internal/integration/rules.go

```diff
@@ -209,38 +209,47 @@ func addAttributes(targetMetrics *TargetMetrics, rules []AddAttributesRule) {
 type ignoreRules []IgnoreRule
 
 func (rules ignoreRules) shouldIgnore(name string, metricType metricType) bool {
-	var prefixesLen, metricTypesLen, exceptRulesLen int
+	if rules.isMetricExcludedFromFilters(name) {
+		return false
+	}
+
 	for _, rule := range rules {
-		exceptRulesLen += len(rule.Except)
-		for _, prefix := range rule.Except {
-			if strings.HasPrefix(name, prefix) {
-				return false
-			}
+		// if metricTypesLen or prefixesLen are not defined and exceptRulesLen
+		// is not empty then all not excepted metric should be dropped
+		totalDroppingRules := len(rule.MetricTypes) + len(rule.Prefixes)
+		if totalDroppingRules == 0 && len(rule.Except) != 0 {
+			return true
 		}
 
 		// MetricTypes
-		metricTypesLen += len(rule.MetricTypes)
 		for _, rMetricType := range rule.MetricTypes {
 			if strings.EqualFold(rMetricType, string(metricType)) {
 				return true
 			}
 		}
 
 		// Prefixes
-		prefixesLen += len(rule.Prefixes)
 		for _, prefix := range rule.Prefixes {
 			if strings.HasPrefix(name, prefix) {
 				return true
 			}
 		}
 	}
 
-	if prefixesLen > 0 || metricTypesLen > 0 {
-		return false
+	return false
+}
+
+// When matching an except rule we do not drop the metric, no matter if a rule is dropping it after
+func (rules ignoreRules) isMetricExcludedFromFilters(name string) bool {
```

Comment:
nit: I would change the terms exclude and filter for except and rules. 
`metricExcepted` or `metricExceptedByRules`

Reply:
true, in the code the field is .except